src/main/io/vtx_rtc6705.c

```diff
@@ -232,6 +232,18 @@ static uint8_t vtxRTC6705GetPowerLevels(const vtxDevice_t *vtxDevice, uint16_t *
     return 0;
 }
 
+static void vtxRTC6705SerializeDeviceStatus(const vtxDevice_t *vtxDevice, sbuf_t *dst)
+{
+    const vtxDevType_e vtxType = vtxCommonGetDeviceType(vtxDevice);
+    const bool deviceReady = vtxCommonDeviceIsReady(vtxDevice);
+    const uint8_t vtxDeviceStatusSize = 2;
+
+    sbufWriteU8(dst, vtxDeviceStatusSize);
+    sbufWriteU8(dst, vtxType);
```

Comment:
Actually, looking at this, we can simplify this a bit and avoid the 'empty' implementations for Tramp / RTC6705 for now if we change the format of the MSP response a bit:

```
uint8_t vtxType // common for all
unit8_t deviceReady // common for all
uint8_t customStatusSize // the output of 'vtxCommonSerializeDeviceStatus()' starts here
[...]
```

In this case the code in `msp.c` can be:

```
            const vtxDevice_t *vtxDevice = vtxCommonDevice();
￼            const vtxDevType_e vtxType = vtxCommonGetDeviceType(vtxDevice);
￼            const bool deviceReady = vtxCommonDeviceIsReady(vtxDevice);
￼
￼            sbufWriteU8(dst, vtxDeviceStatusSize);
￼            sbufWriteU8(dst, vtxType);

            if (!vtxCommonSerializeDeviceStatus(vtxDevice, dst)) {
                sbufWriteU8(dst, 0);
            }
```

(with `vtxCommonSerializeDeviceStatus()` returning `false` if the function pointer is NULL)
￼

Reply:
oh yeah cool make sense!

Reply:
oh yeah cool, make sense!

Reply:
In the whole story here I am still a little bit confused why we even need to send the data size. Configurator must know how to deserialize different vtx_device_status objects anyway.

Reply:
In general MSP's concept of extensibility is based on only ever _appending_ new fields _to the end of a message_. Backwards compatibility is achieved by every party reading the entire message, but then only interpreting as much of it as it knows about, ignoring any additional bytes at the end.
So if we are ever to add more common fields to this message, we will have to do so _after_ the device specific ones, in order to retain backwards compatibility. And for the receiver to know where the next common field is it has to know how long the block of custom fields is.